torch/distributed/rpc.py

```diff
@@ -139,8 +140,16 @@ def remote(to, func, args=None, kwargs=None):
     args = args if args else ()
     kwargs = kwargs if kwargs else {}
 
-    return invoke_remote_builtin(
-        _agent, _to_worker_id(to), qualified_name, *args, **kwargs)
+    if qualified_name is not None:
+        return invoke_remote_builtin(
+            _agent, _to_worker_id(to), qualified_name, *args, **kwargs)
+    else:
+        to = _to_worker_id(to)
+        set_current_rpc_dst(to)
+        rref = invoke_remote_python_udf(
+            _agent, to, serialize(PythonUDF(func, args, kwargs)))
```

Comment:
Suggestion: have serialize take "dst" as argument and deal with "set_current_rpc_dst" inside of it. This will make sure that all calls to "serialize" pass "dst", which is required for all python object serialization within the RPC system (including the serialization of the result of python calls, or the serialization of the return value of to_here as well)